### PR TITLE
chore(Communities): Refactor amounts handling for displaying, minting, airdropping and burning

### PR DESCRIPTION
**Pull request by micieslak**
### What does the PR do

This PR refactors functionality of minting, airdrop and burn regarding handling fees. Provided earlier utilities `AmountsArithmetic` are used to handle amounts as big integers instead of floats. It prevents from problems described in https://github.com/status-im/status-desktop/issues/11376. Permissions functionality will be covered in a separate task.

Closes: #11491

Permissions part is going to be covered in a separate task: https://github.com/status-im/status-desktop/issues/11863

### Affected areas
Multiple components for community settings, `AmountsArithmetic`

### CODE REVIEW

**Review by alexandraB99 — APPROVED** [review]
haven't tested - code looks good!

File: ui/app/AppLayouts/Communities/views/EditCommunityTokenView.qml [hunk 1]
```diff
@@ -264,14 +270,13 @@ StatusScrollView {
...
+                const supplyNumber = parseInt(text)
+                if (Number.isNaN(supplyNumber) || Object.values(errors).length)
                     return
 
-                if(root.isAssetView)
-                    asset.supply = amount
-                else
-                    collectible.supply = amount
+                const token = root.isAssetView ? root.asset : root.collectible
+                token.supply = SQUtils.AmountsArithmetic.fromNumber(
```
Comment (caybro) [inline ui/app/AppLayouts/Communities/views/EditCommunityTokenView.qml:278]: This looks wrong; the old code would assign to toplevel props (`asset` or `collectible`) but the new one just modifies a local variable?

**Review by caybro — COMMENTED** [review]
There are other places which imho need some fixes (like `WalletAccountsModel` in storybook)?

File: ui/app/AppLayouts/Communities/views/EditCommunityTokenView.qml (hunk 1 above)
Comment (micieslak) [reply ui/app/AppLayouts/Communities/views/EditCommunityTokenView.qml:278]: no, `token` is a reference, it works exactly as before

**Comment by micieslak** [comment]
> There are other places which imho need some fixes (like `WalletAccountsModel` in storybook)?

The functionality I've refactored doesn't use wallte model's assets, where amounts are expressed as floats. That should be changed in the next iteration when functionality using that will be altered.

**Review by caybro — APPROVED** [review]
LGTM

**Comment by glitchminer** [comment]
Hey @micieslak , not entirely sure if this is covered by these changes but there was a discrepancy noted when viewing remaining tokens and attempting to airdrop.

1. Create token with `999,999,999` supply and `10` decimal places
2. Airdrop `0.000000001`
3. Remaining supply still shows `999,999,999`
image
5. Start another airdrop - `999,999,999` is accepted without error despite the available balance being only `999999998.999999999`
image

6. Also note that despite being configured with 10 decimal places the number that is able to be used while airdropping depends on the size of integer
image

**Comment by micieslak** [comment]
> Hey @micieslak , not entirely sure if this is covered by these changes but there was a discrepancy noted when viewing remaining tokens and attempting to airdrop.

Thanks @glitchminer for pointing those issues. The same problem exists in the current master and they are not directly related to this PR so I created separate tasks for them:

https://github.com/status-im/status-desktop/issues/11917
https://github.com/status-im/status-desktop/issues/11918

**Review by endulab — APPROVED** [review]
LGTM

**Pull request merged**